crates/iceberg/src/spec/values.rs

```diff
@@ -996,10 +1039,552 @@ mod timestamptz {
+                RawLiteralEnum::Float(v) => match ty {
+                    Type::Primitive(PrimitiveType::Float) => Ok(Some(Literal::float(v))),
+                    _ => Err(invalid_err("float")),
+                },
+                RawLiteralEnum::Double(v) => match ty {
+                    Type::Primitive(PrimitiveType::Double) => Ok(Some(Literal::double(v))),
+                    _ => Err(invalid_err("double")),
+                },
+                RawLiteralEnum::String(v) => match ty {
+                    Type::Primitive(PrimitiveType::String) => Ok(Some(Literal::string(v))),
+                    _ => Err(invalid_err("string")),
+                },
+                // # TODO
+                // rust avro don't support deserialize any bytes representation now.
+                RawLiteralEnum::Bytes(_) => Err(invalid_err("bytes")),
+                RawLiteralEnum::List(v) => match ty {
+                    Type::List(ty) => Ok(Some(Literal::List(
+                        v.list
+                            .into_iter()
+                            .map(|v| {
+                                if let Some(v) = v {
+                                    v.try_into(&ty.element_field.field_type)
+                                } else {
+                                    Ok(None)
+                                }
+                            })
+                            .collect::<Result<_, Error>>()?,
+                    ))),
+                    Type::Map(map_ty) => {
+                        let key_ty = map_ty.key_field.field_type.as_ref();
+                        let value_ty = map_ty.value_field.field_type.as_ref();
+                        let mut map = BTreeMap::new();
+                        for k_v in v.list {
+                            // In deserialize, the element always be `Som`e. `None` will be represented
+                            // as `Some(RawLiteral::Null)`
+                            let k_v = k_v.ok_or_else(|| invalid_err("list"))?;
+                            if let RawLiteralEnum::Record(Record {
+                                required,
+                                optional: _,
+                            }) = k_v
+                            {
+                                if required.len() != 2 {
+                                    return Err(invalid_err("list"));
+                                }
+                                let mut key = None;
+                                let mut value = None;
+                                required.into_iter().for_each(|(k, v)| {
+                                    if k == "key" {
+                                        key = Some(v);
+                                    } else if k == "value" {
+                                        value = Some(v);
+                                    }
+                                });
+                                match (key, value) {
+                                    (Some(k), Some(v)) => {
```

Comment:
I think whether value could be `None` depends on `map_value_field.required`?

Reply:
[The deserialize for `Value::Null` is call `visit_unit()`](https://github.com/apache/avro/blob/369ae568f1017b54fc00bd2f49fd2ebe94fb1b2e/lang/rust/avro/src/de.rs#L252) and we directly convert to `RawLiteral::Null`. So we always represent `None` as `Some(RawLiteral::Null)`, in other words, every `RawLiteral` is `Some(T)` in deserialize. And we will convert `RawLiteral::Null` to `None` in `RawLiteral.try_into(ty) -> Option<Literal>`.

But we indeed need to check whether `Option<Literal>` is `Some(Literal)` when `map_value_field.required=true` here. I will add it.